experimental/dds/tree2/src/feature-libraries/forestSummarizer.ts

```diff
@@ -47,12 +50,16 @@ export class ForestSummarizer implements Summarizable {
 	 * @returns a snapshot of the forest's tree as a string.
 	 */
 	private getTreeString(stringify: SummaryElementStringifier): string {
-		// TODO: maybe assert there are no other roots
-		// (since we don't save them, and they should not exist outside transactions).
-		moveToDetachedField(this.forest, this.cursor);
-		const roots = mapCursorField(this.cursor, jsonableTreeFromCursor);
+		const fields: [string, JsonableTree[]][] = [];
+		this.forest.getDetachedFields().forEach((detachedField) => {
+			moveToDetachedField(this.forest, this.cursor, detachedField);
+			fields.push([
+				this.cursor.getFieldKey(),
+				mapCursorField(this.cursor, jsonableTreeFromCursor),
+			]);
+		});
 		this.cursor.clear();
-		return stringify(roots);
+		return stringify(fields);
```

Comment:
since we are changing the encoded format here, it would be nice to bump the format version number relevent to this string so that people opening old docs get a nice error saying their format is unsupported rather than data corruption/random crashes. I do hope we have such a version number somewhere.

Prabably also include a changeset with this PR noting that it breaks all existing documents.

Reply:
Turns out version stamping is gonna be addressed in this PR: https://github.com/microsoft/FluidFramework/pull/17394/files

Reply:
That should only handel versioning of the formant in which the tree is encoded.

You are changing what tree you put in the summary and changing the path so it starts one field higher. This is a logical change to the forest summarizer format, just the format of the actual encoded trees. Another way to think about it: each bit of code which crates persisted data needs a version. The link you provide solves the fact that jsonableTreeFromCursor doesn't do versioned formatting. Regardless the logic in this file (such as the policy to stringify it, list the files as array of [key, jsonable tree] etc, needs versioning.

This could be covered by a version for just the content in this file, or top level summary version. Bumping either of those should be sufficient. If we don't have either of those, thats something we really need to fix.

Reply:
@Abe27342 would know more but sounds like we don't have anything right now. That PR is the first of the changes planned to fix this.